pydm/application.py

```diff
@@ -115,12 +117,14 @@ def __init__(self, ui_file=None, command_line_args=[], display_args=[], perfmon=
         self.hide_menu_bar = hide_menu_bar
         self.hide_status_bar = hide_status_bar
         self.__read_only = read_only
-        # Open a window if one was provided.
+        # Open a window if required.
         if ui_file is not None:
+            self.make_main_window()
             self.make_window(ui_file, macros, command_line_args)
-            self.had_file = True
-        else:
-            self.had_file = False
+        elif use_main_window:
+            self.make_main_window()
+
```

Comment:
You can simplify this as:

```python
if use_main_window:
    self.make_main_window()

if ui_file is not None:
    self.make_window(ui_file, macros, command_line_args)
```

Please also add the `use_main_window` to the docstrings.

Reply:
@fernandohds564 you are absolutely right!
Thank you for pointing that to me! Your implementation is really accurate! 👍 
As soon as you send the docstring I will merge the changes.